app/models/couch.rb

```diff
@@ -16,6 +16,10 @@ def self.search(params)
       .distinct
   end
 
+  def available_nights
+    self.nights.map { |night| night.date if night.reservation_id.nil? }
```

Comment:
@ski-climb This method is only being used by the datepicker jquery. The gon.available_nights (gon is a gem to pass data from rails to js) is passing on this comma separated array of date strings to the enableAllTheseDays js function. That function reads the dates in the array and only makes those available for selection in the datepicker. As far as I can tell (or better yet the only way I could figure out to get this functionality in datepicker) is by passing it this array of date strings. But let's definitely fine-tooth comb it in the morning.